src/components/program/ProgramPage.jsx

```diff
@@ -8,45 +6,27 @@ import { ErrorPage } from '@edx/frontend-platform/react';
 
 import { MainContent, Sidebar } from '../layout';
 import { LoadingSpinner } from '../loading-spinner';
-import { ProgramContextProvider } from './ProgramContextProvider';
 import ProgramHeader from './ProgramHeader';
 import ProgramMainContent from './ProgramMainContent';
 import ProgramSidebar from './ProgramSidebar';
 
-import { useAllProgramData } from './data/hooks';
 import ProgramEndorsements from './ProgramEndorsements';
 import ProgramFAQ from './ProgramFAQ';
 import ProgramCTA from './ProgramCTA';
 import NotFoundPage from '../NotFoundPage';
 import { PROGRAM_NOT_FOUND_MESSAGE, PROGRAM_NOT_FOUND_TITLE } from './data/constants';
 import ProgramDataBar from './ProgramDataBar';
-import { useEnterpriseCustomer } from '../app/data';
+import { useEnterpriseCustomer, useProgramDetails } from '../app/data';
 
 const ProgramPage = () => {
-  const { programUuid } = useParams();
   const { data: enterpriseCustomer } = useEnterpriseCustomer();
-
-  const [programData, fetchError] = useAllProgramData({ enterpriseUuid: enterpriseCustomer.uuid, programUuid });
-
-  const initialState = useMemo(
-    () => {
-      if (!programData) {
-        return undefined;
-      }
-      const { programDetails } = programData;
-
-      return {
-        program: programDetails,
-      };
-    },
-    [programData],
-  );
+  const { data: program, isError: fetchError, isLoading } = useProgramDetails();
 
   if (fetchError) {
```

Comment:
If the service method to fetch program details is wrapped with a `try/catch` so it doesn't blow up the app if/when API errors when called in the route loader, there won't be an error bubbled up to the `useQuery` result.

In that case, we should likely either return `NotFoundPage` or let the JS error that would throw when accessing properties on `undefined` (for example) bubble up to the `RouteErrorBoundary` instead of handling it within the UI component itself.